README.md

```diff
@@ -1,707 +1,1799 @@
+  ```
+
+  that just immediately returns the extracted value will never update, because it doesn't see any field accesses to check.
+
+</details>
+
+<details>
+<summary><b>Use Cases</b></summary>
+
+- It is likely best used for cases where you need to access specific nested fields in data, and avoid recalculating if other fields in the same data objects are immutably updated.
+
+</details>
+
+<details><summary><b>Parameters</b></summary>
+
+| Name   | Description                  |
+| :----- | :--------------------------- |
+| `func` | The function to be memoized. |
+
+</details>
+
+<details><summary><b>Returns</b></summary>
+
+A memoized function with a `.clearCache()` method attached.
+
+</details>
+
+<details><summary><b>Type parameters</b></summary>
+
+| Name   | Description                                |
+| :----- | :----------------------------------------- |
+| `Func` | The type of the function that is memoized. |
+
+</details>
+
+<details><summary><b><code>Examples</code></b></summary>
+
+###### Using [`createSelector`]
+
+```ts
+import {
+  unstable_autotrackMemoize as autotrackMemoize,
+  createSelector
+} from 'reselect'
+
+const selectTodoIds = createSelector(
+  [(state: RootState) => state.todos],
+  todos => todos.map(todo => todo.id),
+  { memoize: autotrackMemoize }
+)
+```
+
+###### Using [`createSelectorCreator`]
+
+```ts
+import {
+  unstable_autotrackMemoize as autotrackMemoize,
+  createSelectorCreator
+} from 'reselect'
+
+const createSelectorAutotrack = createSelectorCreator({
+  memoize: autotrackMemoize
+})
+
+const selectTodoIds = createSelectorAutotrack(
+  [(state: RootState) => state.todos],
+  todos => todos.map(todo => todo.id)
+)
+```
+
+- Pros:
+  - It is likely to avoid excess calculations and recalculate fewer times than defaultMemoize will, which may also result in fewer component re-renders.
+- Cons:
+
+  - It only has a cache size of 1.
+  - It is slower than defaultMemoize, because it has to do more work. (How much slower is dependent on the number of accessed fields in a selector, number of calls, frequency of input changes, etc)
+  - It can have some unexpected behavior. Because it tracks nested field accesses, cases where you don't access a field will not recalculate properly. For example, a badly-written selector like:
+
+  ```ts
+  createSelector([state => state.todos], todos => todos)
+  ```
+
+  that just immediately returns the extracted value will never update, because it doesn't see any field accesses to check.
+
+**Use Cases for `autotrackMemoize`:**
+
+- It is likely best used for cases where you need to access specific nested fields in data, and avoid recalculating if other fields in the same data objects are immutably updated.
+
+</details> -->
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+<a id="createselectorcreator" ></a>
+
+### createSelectorCreator(memoize | options, ...memoizeOptions)
```

Comment:
:hand: let's move this up to be directly after `createSelector`.  I think the reasonable ordering is:

- `createSelector`
- `createSelectorCreator`
- `createStructuredSelector`
- memoization functions

Reply:
It was ordered that way initially but you're correct, I will rearrange them.